astropy/units/tests/test_equivalencies.py

```diff
@@ -627,6 +628,58 @@ def test_spectraldensity_not_allowed(from_unit, to_unit):
         to_unit.to(from_unit, 1, u.spectral_density(1 * u.AA))
 
 
+@dataclasses.dataclass(eq=False)
+class DuckQuantity(np.lib.mixins.NDArrayOperatorsMixin):
```

Comment:
I think this should be modified to target the Array API, as the other comment indicates.